scripts/globals/events/starlight_celebrations.lua

```diff
@@ -0,0 +1,943 @@
+    local missedFlee = player:getLocalVar("missedFlee")
+    local seconds = math.fmod(os.time() - playerTime, 60)
+    local minutes = math.floor((os.time() - playerTime) / 60)
+
+    if playerTime ~= 0 then
+        if playerPoint ~= 0 then
+            if (npcPoint == 0 and playerPoint ~= 10) then
+                player:setLocalVar("playerBCCP", playerPoint + 1)
+                player:setLocalVar("Checkpoint" .. npcID, 1)
+                if player:getStatusEffect(xi.effect.FLEE) ~= nil or playerPoint == 0 or missedFlee == 0 then
+                    player:setLocalVar("missedFlee", 0)
+                    player:showText(npc, id.text.SMILEHELPER_CHECKPOINT_2, 0, playerPoint, minutes, seconds)
+                    player:addStatusEffect(xi.effect.FLEE, 100, 0, 30)
+                else
+                    local rnd = math.random(0, 1)
+                    if rnd ~= 3 then
+                        player:setLocalVar("missedFlee", 1)
+                        player:showText(npc, id.text.SMILEHELPER_CHECKPOINT_1, 0, playerPoint, minutes, seconds)
+                    else
+                        player:setLocalVar("missedFlee", 0)
+                        player:showText(npc, id.text.SMILEHELPER_CHECKPOINT_2, 0, playerPoint, minutes, seconds)
+                        player:addStatusEffect(xi.effect.FLEE, 100, 0, 30)
+                    end
+                end
+            elseif playerPoint == 10 then
+                player:showText(npc, id.text.SMILEHELPER_POINTS_CLEARED)
+                player:addStatusEffect(xi.effect.FLEE, 100, 0, 30)
+            elseif npcPoint ~= 0 then
+                player:showText(npc, id.text.SMILEHELPER_ALREADY_VISITED, 0, playerPoint, minutes, seconds)
+            end
+        end
+    else
+        player:showText(npc, ID.text.SMILEHELPER_IDLE)
+    end
+end
+
+----------------------------------------------------------------
+-- Apply Zone Decorations, Vendors, and Additional Event NPCs --
+----------------------------------------------------------------
+function xi.events.starlightCelebration.applyStarlightDecorations(zoneid)
+    if xi.events.starlightCelebration.isStarlightEnabled() ~= 0 then
+        local decoration = zones[zoneid].npc.STARLIGHT_DECORATIONS
+        if decoration then
+            for id, v in pairs(decoration) do
+                local npc = GetNPCByID(id)
+                if npc then
+                    npc:setStatus(xi.status.NORMAL)
+                end
+            end
+        end
+    end
+end
+
+function xi.events.starlightCelebration.toggleSmileHelpers(zoneid)
+    if xi.events.starlightCelebration.isStarlightEnabled() ~= 0 then
+        local smileHelper = zones[zoneid].npc.SMILE_HELPERS
+        if smileHelper then
+            for id, v in pairs(smileHelper) do
+                local npc = GetNPCByID(id)
+                if npc then
+                    npc:setStatus(xi.status.NORMAL)
+                    npc:setLocalVar("depopTime", (os.clock() + 3600))
```

Comment:
Is there a reason for os.clock() over os.time()?

Reply:
Not really, beyond just what I've been using. Is os.time() preferred? Will update!

Reply:
I don't see a lot of os.clock in the code, so I'd imagine there's some reason. Should be interchangeable for your code

Reply:
Swapped over to time where needed on all Starlight functions!